src/plugins/shared_ux/public/components/page_template/no_data_page/no_data_card/elastic_agent_card.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { ElasticAgentCardProps } from './types';
+import { useHttp, usePermissions } from '../../../../services';
+import { ElasticAgentCardComponent } from './elastic_agent_card.component';
+
+export const ElasticAgentCard = (props: ElasticAgentCardProps) => {
+  const { canAccessFleet } = usePermissions();
+  const { addBasePath } = useHttp();
+
+  const createHref = () => {
```

Comment:
Instead of passing the entire `addBasePath` as a prop to the pure component, this component is now responsible for creating href.